androidapp/Connexus/app/src/main/java/com/green/apt/connexus/ViewSingleActivity.java

```diff
@@ -0,0 +1,123 @@
+package com.green.apt.connexus;
+
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.GridView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+import com.green.apt.connexus.R;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ViewSingleActivity extends AppCompatActivity {
+
+    private static final String BASE_URL = "https://apt-miniproject-greenteam-v2.appspot.com/api/";
+    private static String relativeUrl = "view";
+    private static String stream = "";
+    private static String streamName = "Single Stream";
+    private static Long streamId = null;
+    private static List<String> streamPhotos = new ArrayList<>();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_view_single);
+        Intent intent = getIntent();
+        Bundle extras = intent.getExtras();
+        streamName = extras.getString("SINGLE_STREAM_NAME");
+        streamId = Long.parseLong(extras.getString("SINGLE_STREAM_ID"));
+
+        // Capture the layout's TextView and set the string as its text
+        TextView textView = (TextView) findViewById(R.id.textView);
+        textView.setText(streamName);
+        streamPhotos.clear();
+
+        this.getStreamPhotos();
+    }
+
+    public void goToStreams(View view) {
+        Intent intent = new Intent(this, ViewAllActivity.class);
+        startActivity(intent);
+    }
+
+    private void parseStream(){
```

Comment:
I think we can reuse if we move it to the model :)

Reply:
Parse stream is not exactly the same for every activity since our api returns jsons in different formats all over the place not sure how to keep this compact yet variable

Reply:
Yeah, no worries I think we can use lambdas for that, you don't need to make the change.